packages/gatsby/src/redux/reducers/queries.ts

```diff
@@ -195,6 +202,7 @@ export function queriesReducer(
       const { path } = action.payload
       const query = registerQuery(state, path)
       query.dirty = 0
+      query.running = 0 // TODO: also reset the state on pre-bootstrap (as it could've been persisted)
```

Comment:
should this TODO be implemented in this PR? I would think so, because again we could (maybe?) end up waiting for "running" query that doesn't actually run in current session

Reply:
Yup, forgot about this piece. Thanks for noticing!